docs/user-guide/aae-extensions.md

```diff
@@ -0,0 +1,191 @@
+
+    constructor(formRenderingService: FormRenderingService) {
+        this.formRenderingService.register({
+            'text': () => CustomEditorComponent
+        }, true);
+    }
+
+}
+```
+
+At runtime it should look similar to the following:
+
+![custom text widget](../docassets/images/aae-simple-override-form.png)
+
+## Replacing custom form widgets with custom components
+
+This is a short walkthrough on rendering custom AAE form widgets by means of custom Angular components.
+
+### Creating custom form widget
+
+First let's create a basic form widget and call it `Custom Editor`:
+
+![custom form widget](../docassets/images/aae-form-widget.png)
+
+_Note the `type` value as it will become a `field type` value when corresponding form is rendered._
+
+Next put some simple html layout for [`Form`](../../lib/process-services/src/lib/task-list/models/form.model.ts)`runtime template` and [`Form`](../../lib/process-services/src/lib/task-list/models/form.model.ts)`editor template` fields:
+
+```html
+<div style="color: blue">Custom form widget</div>
+```
+
+Now you are ready to design a test form based on your custom form widget:
+
+![custom form widget form](../docassets/images/aae-form-with-widget.png)
+
+### Creating custom widget
+
+Once wired with a new task it should look like the following within AAE Process Experience:
+
+![adf form widget](../docassets/images/aae-unresolved-widget.png)
+
+
+Let's create an Angular component to render missing content:
+
+```ts
+import { Component } from '@angular/core';
+import { WidgetComponent } from '@alfresco/adf-core';
+
+@Component({
+    selector: 'app-demo-widget',
+    template: `<div style="color: green">ADF version of custom form widget</div>`
+})
+export class DemoWidgetComponent extends WidgetComponent {}
+```
+
+Put it inside custom module:
+
+```ts
+import { NgModule } from '@angular/core';
+import { DemoWidgetComponent } from './demo-widget.component';
+
+@NgModule({
+    declarations: [ DemoWidgetComponent ],
+    exports: [ DemoWidgetComponent ]
+})
+export class CustomWidgetsModule {}
+```
+
+And import into your Application Module
+
+```ts
+@NgModule({
+    imports: [
+        // ...
+        CustomWidgetsModule
+        // ...
+    ],
+    providers: [],
+    bootstrap: [ AppComponent ]
+})
+export class AppModule {}
+```
+
+Now you can import [`FormRenderingService`](../core/services/form-rendering.service.md) in any of your Views and provide new mapping:
+
+```ts
+import { Component } from '@angular/core';
+import { DemoWidgetComponent } from './demo-widget.component';
+
+@Component({...})
+export class MyView {
+
+    constructor(formRenderingService: FormRenderingService) {
+        this.formRenderingService.register({
+            'custom-editor': () => DemoWidgetComponent
+        });
+    }
+}
+```
+
+At runtime you should now see your custom Angular component rendered in place of the form widgets:
+
+![adf form widget runtime](../docassets/images/aae-resolved-widget.png)
```

Comment:
```suggestion
To render the missing content:

1. Create an Angular component:

    ```ts
    import { Component } from '@angular/core';
    import { WidgetComponent } from '@alfresco/adf-core';
    @Component({
        selector: 'app-demo-widget',
        template: `<div style="color: green">ADF version of custom form widget</div>`
    })
    export class DemoWidgetComponent extends WidgetComponent {}
    ```

2. Place it inside the custom module:

    ```ts
    import { NgModule } from '@angular/core';
    import { DemoWidgetComponent } from './demo-widget.component';
    @NgModule({
        declarations: [ DemoWidgetComponent ],
        exports: [ DemoWidgetComponent ]
    })
    export class CustomWidgetsModule {}
    ```

3. Import it into your Application Module:

    ```ts
    @NgModule({
        imports: [
            // ...
            CustomWidgetsModule
            // ...
        ],
        providers: [],
        bootstrap: [ AppComponent ]
    })
    export class AppModule {}
    ```

4. Import the [`FormRenderingService`](../core/services/form-rendering.service.md) in any of your Views and provide the new mapping:

    ```ts
    import { Component } from '@angular/core';
    import { DemoWidgetComponent } from './demo-widget.component';
    @Component({...})
    export class MyView {
        constructor(formRenderingService: FormRenderingService) {
            this.formRenderingService.register({
                'custom-editor': () => DemoWidgetComponent
            });
        }
    }
    ```

At runtime you should now see your custom Angular component rendered in place of the original form widgets:

![adf form widget runtime](../docassets/images/aae-resolved-widget.png)
```